packages/openid4vc/src/openid4vc-verifier/OpenId4VcSiopVerifierService.ts

```diff
@@ -608,9 +624,28 @@ export class OpenId4VcSiopVerifierService {
 
         let isValid: boolean
 
-        // TODO: it might be better here to look at the presentation submission to know
-        // If presentation includes a ~, we assume it's an SD-JWT-VC
-        if (typeof encodedPresentation === 'string' && encodedPresentation.includes('~')) {
```

Comment:
let's keep sd-jwt first, as it's cheaper to check for inclusion of `~` than to parse the mdoc.

Also this is what i meant in other comment with check if `isBase64DeviceResponse` and then `MdocDeviceResponse.fromBase64Url`. Might as well save the result from the first call and then using that to not do the whole cbor decoding twice.